### PR TITLE
Bump vm-builder v0.21.0 -> v0.23.2

### PR DESCRIPTION
**Pull request by sharnoff**
Relevant changes were all from v0.23.0:

- neondatabase/autoscaling#724
  - IIRC this requires additional changes to redirect the logs through the socket. cc @Omrigan - happy to do that in this PR or a follow-up. 
- neondatabase/autoscaling#726
- neondatabase/autoscaling#732

cc @neondatabase/autoscaling

---

NOTE: This may or may not work, until neondatabase/cloud#9675 is merged to update cloud CI.

NOTE: Do not merge until after the platform release 2024-01-29 (next Monday). We want this to sit on staging for a while longer.

### CODE REVIEW

**Review by shayanh — APPROVED** [review]

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -867,7 +867,7 @@ jobs:
       run:
         shell: sh -eu {0}
     env:
-      VM_BUILDER_VERSION: v0.21.0
+      VM_BUILDER_VERSION: v0.23.0
```
Comment (bayandin) [inline .github/workflows/build_and_test.yml:870]:
```suggestion
      VM_BUILDER_VERSION: v0.23.1
```

**Comment by Omrigan** [comment]
> IIRC this requires additional changes to redirect the logs through the socket. cc @Omrigan - happy to do that in this PR or a follow-up.

I think this should be done in a separate PR: for now it should keep working with logs in `/dev/console`

**Comment by Omrigan** [comment]
> IIRC this requires additional changes to redirect the logs through the socket. 

I think this should be done in a separate PR: for now it should keep working with logs in `/dev/console`

> @Omrigan - happy to do that in this PR or a follow-up.

You can do this or you can point me where `compute_ctl` log can be configured, and I do it. Happy with whatever!

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -894,7 +894,7 @@ jobs:
       run:
         shell: sh -eu {0}
     env:
-      VM_BUILDER_VERSION: v0.21.0
+      VM_BUILDER_VERSION: v0.23.1
```
Comment (bayandin) [inline .github/workflows/build_and_test.yml:897]:
```suggestion
      VM_BUILDER_VERSION: v0.23.2
```

**Pull request merged**